### DISCUSSION TITLE
Can SegmentTemplate @duration be a non-integer?

### DISCUSSION THREAD

**Discussion by joeyparrish** [Dash-Industry-Forum/DASH-IF-IOP#414] [Q&A] [answered]
In this sample:
  https://dash.akamaized.net/akamai/bbb_30fps/bbb_with_multiple_tiled_thumbnails.mpd

We found this snippet:

```xml
  <AdaptationSet id="3" mimeType="image/jpeg" contentType="image">
    <SegmentTemplate media="$RepresentationID$/tile_$Number$.jpg"
        duration="634.566" startNumber="1"/>
```

Is this sample of out conformance to the spec for having a non-integer `duration` attribute (634.566)?  Should the content be fixed to use something like `duration="634566" timescale="1000"` instead?  Or should players always assume that `duration` could be a non-integer?

Thanks!

**Comment by ojw28** [comment, marked as answer]
The DASH specification defines `duration` to be of type `xs:unsignedInt`, so I think the example violates the spec.

**Comment by haudiobe** [comment]
The answer from @ojw28 is correct. Thanks

**Comment by mikedo** [comment]
FYI, this is easily checked with the XML schema which is authoritative over the text in the standard:
https://github.com/MPEGGroup/DASHSchema/ 
The latest code release matches the forthcoming 5th Ed.  But it includes a good amount of better datatyping in general.
